Src/VimCore/CommonOperations.fs

```diff
@@ -228,10 +228,156 @@ type internal CommonOperations
             TextViewUtil.MoveCaretToPoint _textView firstLine.Start
             _editorOperations.MoveToStartOfLineAfterWhiteSpace(false)
 
-    /// Format the specified line range
-    member x.FormatLines range =
+    /// Format the code lines in the specified line range
+    member x.FormatCodeLines range =
         _vimHost.FormatLines _textView range
 
+        // Place the cursor on the first non-blank character of the first line formatted.
+        let firstLine = SnapshotUtil.GetLine _textView.TextSnapshot range.StartLineNumber
+        TextViewUtil.MoveCaretToPoint _textView firstLine.Start
+        _editorOperations.MoveToStartOfLineAfterWhiteSpace(false)
+
+    /// Format the text lines in the specified line range
+    member x.FormatTextLines (range: SnapshotLineRange) preserveCaretPosition =
+
+        // Get formatting configuration values.
+        let autoIndent = _localSettings.AutoIndent
+        let textWidth =
+            if _localSettings.TextWidth = 0 then
+                79

```

Comment:
Why 79?

Reply:
That's what the vim documentation says it uses if `textwidth` is zero!

> If the 'textwidth' option is 0, the formatted length is the screen width (with a maximum width of 79).

I guess the idea is that it wraps at 80 columns but makes a concession for terminals that "double-wrap" the line if it is exactly 80.

Also, from this documentation you can see that I did not limit the width to the screen width, which doesn't seem to be a reasonable setting to use with Visual Studio where the use has virtually no control over the width of the window.